tests/testthat/test-test-threads.R

```diff
@@ -127,42 +135,47 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
 
     # Dont overwrite NBURN if set to 0
-    max_evals <- get_est_options("MAXEVAL|NITER|NBURN", mods_complex)
+    max_evals <- get_est_options(mods_complex)
 
+    # Confirm that estimation method didnt change, and that MAXEVAL/NITER was preserved
     for(i in seq_along(max_evals)){
-      expect_identical(unname(max_evals[[i]]), c(100, 0, 100))
-      # Confirm that estimation method didnt change, and that MAXEVAL/NITER was preserved
-      expect_identical(names(max_evals[[i]]), c("METHOD=1, MAXEVAL", "METHOD=BAYES, NBURN", "METHOD=BAYES, NITER"))
+      max_evals_i <- unlist(max_evals[[i]])
+      is_nburn_opt <- grep("(?i)nburn", names(max_evals_i))
+      expect_true(all(max_evals_i[is_nburn_opt] == 0))
+      expect_true(length(is_nburn_opt) == 1)
+      expect_true(all(max_evals_i[!is_nburn_opt] == 100))
     }
   })
 
   test_that("test_threads(.dry_run=T) correctly changes maxeval/niter: nburn is handled correctly [BBR-TSTT-002]", {
 
     expect_message(
       mods_complex2 <- test_threads(mod_complex2, .threads = c(2, 4), .cap_iterations = 100, .mode = "local", .dry_run = TRUE),
-      "Adding NBURN declaration"
+      "Adding NBURN"
     )
 
     # Overwrite NBURN since was set to value other than 0
-    max_evals <- get_est_options("MAXEVAL|NITER|NBURN", mods_complex2)
+    max_evals <- get_est_options(mods_complex2)
 
     for(i in seq_along(max_evals)){
-      expect_identical(unname(max_evals[[i]]), c(100, 100, 100, 100))
+      max_evals_i <- unlist(unname(max_evals[[i]]))
+      expect_identical(unname(max_evals_i), c(100, 100, 100, 100))
       # Confirm that estimation method didnt change, and that MAXEVAL/NITER was preserved
-      # This also confirms that NBURN is added when only NITER is specified (recursive test)
-      expect_identical(names(max_evals[[i]]), c("METHOD=SAEM, NBURN", "METHOD=SAEM, NITER", "METHOD=IMP, NITER", "METHOD=IMP, NBURN"))
+      # This also confirms that NBURN is added when only NITER is specified (EST record 2) (recursive test)
+      expect_identical(names(max_evals_i), c("NITER", "NBURN", "NITER", "NBURN"))
     }
   })
 
   test_that("test_threads(.dry_run=T) correctly changes maxeval/niter: keeps original if .cap_iterations = NULL [BBR-TSTT-002]", {
 
     mods_complex <- test_threads(mod_complex, .threads = c(2, 4), .cap_iterations = NULL, .mode = "local", .dry_run = TRUE)
 
-    max_evals <- get_est_options("MAXEVAL|NITER|NBURN", mods_complex)
+    max_evals <- get_est_options(mods_complex)
 
     for(i in 1:length(max_evals)){
-      expect_equal(unname(max_evals[[i]]), c(9999, 0, 10))
-      expect_equal(names(max_evals[[i]]), c("METHOD=1, MAXEVAL", "METHOD=BAYES, NBURN", "METHOD=BAYES, NITER"))
+      max_evals_i <- unlist(unname(max_evals[[i]]))
+      expect_equal(unname(max_evals_i), c(9999, 10, 0))
+      expect_equal(names(max_evals_i), c("MAXEVAL", "NITER", "NBURN"))
```

Comment:
Hmm, there aren't two "BAYES" methods in acop-fake-bayes.ctl:

<https://github.com/metrumresearchgroup/bbr/blob/feddd576f3f8cf9f091487b761391dedba481040/inst/model/nonmem/complex/acop-fake-bayes.ctl#L38-L40>

So I guess `get_est_options` *before* this PR was returning an incorrect result. Is that related to why the 0 and 10 swap places in the expected value?

Reply:
No it was correct before, but was formatted differently and IMO more confusingly. It would attach the method to each of the options, so both of those expectations corresponded to the same "BAYES" method. The flipping of the 10 and 0 is just because the order of the options changed

Reply:
Hmm, I can't make heads or tails of it with a quick glance, but I'll take you word for it.  Thanks.